test/ProjectTest.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Phing\Test;
+
+
+use Phing\Project;
+
+class ProjectTest extends \PHPUnit_Framework_TestCase
+{
+
+    /** @var Project */
+    protected $project;
+
+    protected function setUp()
+    {
+        $this->project = new Project();
+    }
+
+    public function testSettingAndGettingSimpleProperty()
+    {
+        $this->project->setProperty('foo', 'bar');
+        $this->assertEquals('bar', $this->project->getProperty('foo'));
+    }
+
+    public function testPropertyGetWillExpandPropertyValues()
+    {
+        $this->project->setProperty('foo', 'foo');
+        $this->project->setProperty('test', '${foo}bar');
+
+        $this->assertEquals('foobar', $this->project->getProperty('test'));
+
+    }
+
+    public function testGetProperties()
+    {
+        $this->project->setProperty('foo', 'foo');
+        $this->project->setProperty('bar', 'bar');
+
+        $this->assertEquals(array('foo' => 'foo', 'bar' => 'bar'), $this->project->getProperties());
+    }
+
+    public function testGetPropertiesWillExpandPropertyValues()
+    {
+        $this->project->setProperty('foo', 'foo');
+        $this->project->setProperty('bar', '${foo}bar');
+
+        $this->assertEquals(array('foo' => 'foo', 'bar' => 'foobar'), $this->project->getProperties());
```

Comment:
@mrook Surprisingly this does not work ATM. You think it's a glitch or by design?